File: Java_Practice/C/array_sorted.c
#include <stdio.h>

int main() {
    int n, i;
    printf("Enter size of array: ");
    scanf("%d", &n);

    int arr[n];
    printf("Enter %d elements: ", n);
    for (i = 0; i < n; i++) {
        scanf("%d", &arr[i]);
    }

    int isAsc = 1, isDesc = 1;

    for (i = 0; i < n - 1; i++) {
        if (arr[i] > arr[i + 1])
            isAsc = 0;
        if (arr[i] < arr[i + 1])
            isDesc = 0;
    }

    if (isAsc)
        printf("Array is sorted in ascending order.\n");
    else if (isDesc)
        printf("Array is sorted in descending order.\n");
    else
        printf("Array is NOT sorted.\n");

    return 0;
}

File: Java_Practice/Java/array_sorted.java
//Check if Array is Sorted (ascending/descending)
import java.util.Scanner;

public class array_sorted {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter size of array: ");
        int n = sc.nextInt();
        int[] arr = new int[n];
        
        System.out.println("Enter elements:");
        for(int i=0;i<n;i++) arr[i] = sc.nextInt();

        boolean isAsc = true, isDesc = true;
        for(int i=0;i<n-1;i++){
            if(arr[i] > arr[i+1]) isAsc = false;
            if(arr[i] < arr[i+1]) isDesc = false;
        }

        if(isAsc) System.out.println("Array is sorted ascending.");
        else if(isDesc) System.out.println("Array is sorted descending.");
        else System.out.println("Array is NOT sorted.");
    }
}
